examples/Thermostat/Server/ThermoFacade.cs

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) ZeroC, Inc.
+
+using IceRpc;
+using IceRpc.Features;
+using IceRpc.Slice;
+using Igloo;
+using System.Runtime.CompilerServices;
+using System.Threading.Channels;
+
+namespace ThermostatServer;
+
+/// <summary>Implements Slice interface `Thermostat` by forwarding calls to the device or by returning data reported
+/// by the device.</summary>
+/// <remarks>Most of the server-side logic is implemented in this class.</remarks>
+internal sealed class ThermoFacade : Service, IThermostatService
+{
+    private readonly LinkedList<ChannelWriter<Reading>> _channelWriters = new();
+
+    private Reading? _latestReading;
+
+    // Protects all read-write fields.
+    private readonly object _mutex = new();
+
+    private CancellationTokenSource? _publishCts;
+
+    private readonly CancellationToken _shutdownToken;
+
+    private readonly IThermoControl _thermoControl;
+
+    /// <summary>Changes the target temperature by forwarding the call to the device.</summary>
+    public async ValueTask ChangeSetPointAsync(
+        float setPoint,
+        IFeatureCollection features,
+        CancellationToken cancellationToken)
+    {
+        try
+        {
+            await _thermoControl.ChangeSetPointAsync(setPoint, cancellationToken: cancellationToken);
```

Comment:
I think it is uncommon to use an argument label in this case:

```suggestion
            await _thermoControl.ChangeSetPointAsync(setPoint, cancellationToken);
```

Reply:
How does this work? We have a features parameter in between.

Reply:
You are right, we need the label